peps/pep-0790.rst

```diff
@@ -0,0 +1,137 @@
+
+
+Motivation
+==========
+
+The current syntax for catching multiple exceptions requires parentheses in the ``except`` expression: ::
+
+    try:
+        ...
+    except (ExceptionA, ExceptionB, ExceptionC):
+        ...
+
+While this syntax is clear and unambiguous, it can be seen as unnecessarily verbose in some cases, especially when catching a large number of exceptions. By allowing the omission of parentheses, we can simplify the syntax: ::
+
+    try:
+        ...
+    except ExceptionA, ExceptionB, ExceptionC:
+        ...
+
+This change would bring the syntax more in line with other comma-separated lists in Python, such as function arguments, generator expressions inside of a function call, and tuple literals, where parentheses are optional.
+
+
+Rationale
+=========
+
+The decision to allow unparenthesized ``except`` blocks is based on the following considerations:
+
+1. Simplicity: Removing the requirement for parentheses simplifies the syntax, making it more consistent with other parts of the language.
+
+2. Readability: In cases where many exceptions are being caught, the removal of parentheses can improve readability by reducing visual clutter.
+
+3. Consistency: This change makes the ``except`` clause more consistent with other parts of Python where unambiguous, comma-separated lists don't require parentheses.
+
+Specification
+=============
+
+The syntax for the except clause will be modified to allow an unparenthesized list of exception types. The grammar will be updated as follows: ::
+
+    except_block[excepthandler_ty]:
+        | invalid_except_stmt_indent
+        | 'except' e=expressions t=['as' z=NAME { z }] ':' b=block {
+            _PyAST_ExceptHandler(e, (t) ? ((expr_ty) t)->v.Name.id : NULL, b, EXTRA) }
+        | 'except' ':' b=block { _PyAST_ExceptHandler(NULL, NULL, b, EXTRA) }
+        | invalid_except_stmt
+    except_star_block[excepthandler_ty]:
+        | invalid_except_star_stmt_indent
+        | 'except' '*' e=expressions t=['as' z=NAME { z }] ':' b=block {
+            _PyAST_ExceptHandler(e, (t) ? ((expr_ty) t)->v.Name.id : NULL, b, EXTRA) }
+        | invalid_except_star_stmt
+
+
+This allows both the current parenthesized syntax and the new unparenthesized syntax: ::
+
+    try:
+        ...
+    except (ExceptionA, ExceptionB):  # Still valid
+        ...
+    except ExceptionC, ExceptionD:    # New syntax
+        ...
+
+The semantics of exception handling remain unchanged. The interpreter will catch any of the listed exceptions, regardless of whether they are parenthesized or not.
+
+
+Backwards Compatibility
+=======================
+
+This change is fully backwards compatible. All existing code using parenthesized ``except`` blocks will continue to work without modification. The new syntax is purely additive and does not break any existing code.
```

Comment:
It's incompatible with Python 2 though: `except ExceptionA, ExceptionB:` meant one thing in Python 2, and now will mean something entirely different in Python 3. You can dismiss that because nobody should be using Python 2 any more, but I feel it's worth at least discussing.

Reply:
> You can dismiss that because nobody should be using Python 2 any more, but I feel it's worth at least discussing.

I am happy mentioning this. WDYT @brettcannon ?

Reply:
Sure, it's fine to mention.